adapters/YahooAdapter/YahooAdapter/Native/GADMAdapterYahooNativeAd.m

```diff
@@ -0,0 +1,345 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "GADMAdapterYahooNativeAd.h"
+#import <YahooAds/YahooAds.h>
+#import "GADMAdapterYahooBaseClass.h"
+#import "GADMAdapterYahooConstants.h"
+#import "GADMAdapterYahooUtils.h"
+
+@interface GADMAdapterYahooNativeAd () <YASNativeAdDelegate>
+@end
+
+@implementation GADMAdapterYahooNativeAd {
+  /// Yahoo native ad.
+  YASNativeAd *_nativeAd;
+
+  /// Connector from the Google Mobile Ads SDK to receive ad configurations.
+  __weak id<GADMAdNetworkConnector> _connector;
+
+  /// Adapter for receiving ad request notifications.
+  __weak id<GADMAdNetworkAdapter> _adapter;
+
+  /// Placement ID string used to request ads from Verizon Ads SDK.
+  NSString *_placementID;
+
+  /// Yahoo Mobile SDK video player view object.
+  YASVideoPlayerView *_yahooVideoPlayerView;
+}
+
+- (nonnull instancetype)initWithGADMAdNetworkConnector:(nonnull id<GADMAdNetworkConnector>)connector
+                              withGADMAdNetworkAdapter:(nonnull id<GADMAdNetworkAdapter>)adapter {
+  self = [super init];
+  if (self) {
+    _connector = connector;
+    _adapter = adapter;
+
+    NSDictionary<NSString *, id> *credentials = [connector credentials];
+    if (credentials[GADMAdapterYahooPosition]) {
+      _placementID = credentials[GADMAdapterYahooPosition];
+    }
+    NSString *siteID = credentials[GADMAdapterYahooDCN];
+    GADMAdapterYahooInitializeYASAdsWithSiteID(siteID);
+  }
+
+  return self;
+}
+
+- (void)loadNativeAdWithAdTypes:(nonnull NSArray<GADAdLoaderAdType> *)adTypes
```

Comment:
I have some questions regarding native ads - while our test ad loads, I don't see click handling as I would expect, I will investigate further.

Please see below for a couple additional comments & questions.

Reply:
We only have test placements for the `simpleImage` format. Can you confirm that the adapter also works as expected for `simpleVideo` format?